.github/workflows/preview.yml

```diff
@@ -0,0 +1,85 @@
+name: preview
+
+on:
+  workflow_call: ~
+
+permissions:
+  id-token: write
+  pull-requests: write
+  deployments: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Deployment
+        uses: actions/github-script@v7
+        id: deployment
+        with:
+          result-encoding: string
+          script: |
+            const { owner, repo } = context.repo;
+            const deployment = await github.rest.repos.createDeployment({
+                issue_number: context.issue.number,
+                owner,
+                repo,
+                ref: context.payload.pull_request.head.ref,
+                environment: `preview-${context.issue.number}`,
+                description: `Preview deployment for PR ${context.issue.number}`,
+                auto_merge: false,
+                required_contexts: [],
+            })
+            await github.rest.repos.createDeploymentStatus({
+                deployment_id: deployment.data.id,
+                owner,
+                repo,
+                state: "in_progress",
+                description: "Deployment created",
+                log_url: `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}?pr=${context.issue.number}`,
+            })
+            return deployment.data.id
+
+      - uses: actions/checkout@v4
+        
+      - uses: actions/download-artifact@v4
+        with:
+          name: docs
+          path: .artifacts/docs/html
+
+      - uses: ./.github/actions/aws-auth
```

Comment:
How does this work when called from other elastic repositories? 

I'd prefer this to be an action since we don't want to polute other UI's with all these steps (correct me if im wrong) but then auth might be an issue?

Reply:
Sorry, I should have been clearer in my description.

This is implementation is not supposed to be used for the consumer, only for this repo.

I was planning to create a consumer action in a follow-up.

Ofcourse it will be highly inspired by this, and we will see if there can be any commons that we can refactor out.

Reply:
SGTM!